ssl/record/ssl3_record.c

```diff
@@ -217,7 +217,7 @@ int ssl3_get_record(SSL *s)
                                num_recs == 0 ? 1 : 0, &n);
             if (rret <= 0) {
 #ifndef OPENSSL_NO_KTLS
-                if (!BIO_get_ktls_recv(s->rbio) || rret == 0)
+                if (!using_ktls || rret == 0)
```

Comment:
Not strictly a refactor as the definition is different:

https://github.com/openssl/openssl/pull/18304/files#diff-a4131b6fe8ab7af35f45112ef26d47767c8b35f76466c1468e8cad0f381d1ef8R203-R204

This could potentially be a bugfix. In any case, approving.